### ISSUE TITLE
Variables containing functions are not documented

### ISSUE THREAD

**Issue by susnux** [mitmproxy/pdoc#320]
#### Problem Description
If you assign functions to a variable they do not get documented. Example:

```python
# a/__init__.py
from .b import a

b = a("b")
c = a("c")
"""This is c"""
d = a("d")
d.__doc__ = "This is d"

#a/b.py
def a(name):
    """This is a decorator"""
    def wrapper(f):
        """This is the wrapper"""
        print(name)
        return f
    return wrapper

```

So you have a decorator called `a` with parameters and `b`, `c` and `d` as alias decorators (`@a("b")` is the same as doing `@b`).

I expect pdoc to document those "variable" / functions, but the result is that just `c` got documented, but with wrong `__doc__`.
Instead of `This is c` it shows `This is the wrapper`, also the source code shown is from the wrapper.

`b` and `d` vanished at all from the documentation, the docstring is correctly set in python, as you can verify with:
```python
import a
print(a.d.__doc__)
# result: "This is d"
```

#### Steps to reproduce the behavior:
See example above

#### System Information
```pdoc: 8.0.1
Python: 3.9.9
Platform: Linux-5.15.8-1-default-x86_64-with-glibc2.34```

**Comment by susnux** [comment]
Example output
![grafik](https://user-images.githubusercontent.com/1855448/147140287-57c22003-871b-43df-b3ae-4cdb91e9fe23.png)

**Comment by mhils** [comment]
Thanks for the detailed example!

> I expect pdoc to document those "variable" / functions, but the result is that just `c` got documented, but with wrong `__doc__`. Instead of `This is c` it shows `This is the wrapper`, also the source code shown is from the wrapper.

First off I agree that we should show the overridden docstring here, I'll push a fix momentarily. Thanks for pointing that out.

The rest is a bit more tricky. `b` and `d` are just function objects created at runtime. Using dynamic analysis, we can't distinguish whether they have been created in `__init__.py`, or whether they have just been imported from `b.py`. We don't want to document everything that is imported of course. Unfortunately their qualname and module attributes do in fact point to `b.py`:
```python
>>> a.b.__qualname__
'a.b.<locals>.wrapper'
>>> a.b.__module__
'a.b'
```

Long story short, to show up, we either need `__all__` or a manual docstring annotation as done with `c`. 

I'm afraid we also can't do much about the source code. It's showing what `inspect.getsource` is providing us with, function objects created  at runtime is really nothing which can be solved with dynamic analysis.

**Comment by susnux** [comment]
Sounds reasonable, especially because that the docstring of `c` was not overridden confused me the most.
